File: scala/worksheet/src/org/jetbrains/plugins/scala/worksheet/WorksheetLanguage.java
package org.jetbrains.plugins.scala.worksheet;

import com.intellij.lang.Language;
import org.jetbrains.plugins.scala.ScalaLanguage;

public class WorksheetLanguage extends Language {

    public static final WorksheetLanguage INSTANCE = new WorksheetLanguage();

    private WorksheetLanguage() {
        super(ScalaLanguage.INSTANCE, "Scala Worksheet");
    }
}

File: scala/worksheet/src/org/jetbrains/plugins/scala/worksheet/WorksheetLanguage.scala
package org.jetbrains.plugins.scala.worksheet

import com.intellij.lang.Language
import org.jetbrains.plugins.scala.ScalaLanguage

final class WorksheetLanguage private()
  extends Language(ScalaLanguage.INSTANCE, "Scala Worksheet")
    with WorksheetLanguageLike

object WorksheetLanguage {
  final val INSTANCE = new WorksheetLanguage
}